command/agent/config.go

```diff
@@ -1371,6 +1376,9 @@ func MergeConfig(a, b *Config) *Config {
 	if b.NodeID != "" {
 		result.NodeID = b.NodeID
 	}
+	if b.DisableHostNodeID == true {
+		result.DisableHostNodeID = b.DisableHostNodeID
```

Comment:
Oh nice I will check it out. Is the framework done and just needs to be integrated?